packages/hooks/src/useIdAllocator.ts

```diff
@@ -1,53 +1,44 @@
-import { useEffect, useState } from 'react';
-
-import useIsomorphicLayoutEffect from './useIsomorphicLayoutEffect';
-
-let serverHandoffComplete = false;
-const registry: Map<string, number> = new Map();
-
-const genId = (prefix: string) => {
-  if (registry.has(prefix)) {
-    const val = registry.get(prefix);
-    const update = val ? val + 1 : 1;
-    registry.set(prefix, update);
-    return update;
-  } else {
-    registry.set(prefix, 1);
-    return 1;
-  }
-};
-
-/**
- * Generates an SSR-compatible unique id based on a prefix string and an optional idProp parameter.
- * @param options {prefix: string; id: string;}
- * @param options.prefix string that prefixes the generated id
- * @param options.id string that represents an already-created id
- */
-function useIdAllocator({
-  prefix,
-  id: idProp,
-}: {
+// Currently using Material UI useId hook until we can upgrade to React 18's useId
+// https://github.com/mui/material-ui/blob/master/packages/mui-utils/src/useId.ts
+import React, { useEffect, useState } from 'react';
```

Comment:
nit: can we move this to its own `useIdAllocator/` directory, with its own `useIdAllocator.spec.ts` file ?